src/extensions/agent_modules/shared/enrollment_utils/src/adu_enrollment_utils.c

```diff
@@ -0,0 +1,126 @@
+    {
+        Log_Error("Null input (context:%p, data:%p)", context, context->data);
+        return NULL;
+    }
+
+    return (ADUC_Enrollment_Request_Operation_Data*)(context->data);
+}
+
+/**
+ * @brief Sets the enrollment state and updates 'IsDeviceEnrolled' state in the DU agent state store.
+ * @param data The enrollment data object.
+ * @param state The enrollment state to set.
+ * @return ADU_ENROLLMENT_STATE The old enrollment state.
+ */
+ADU_ENROLLMENT_STATE EnrollmentData_SetState(
+    ADUC_Enrollment_Request_Operation_Data* enrollmentData, ADU_ENROLLMENT_STATE state, const char* reason)
+{
+    ADU_ENROLLMENT_STATE oldState = enrollmentData->enrollmentState;
+    if (enrollmentData->enrollmentState != state)
+    {
+        Log_Info(
+            "Enrollment state changed from %d to %d (reason:%s)",
+            enrollmentData->enrollmentState,
+            state,
+            IsNullOrEmpty(reason) ? "unknown" : reason);
+        enrollmentData->enrollmentState = state;
+
+        if (ADUC_StateStore_SetIsDeviceEnrolled(state == ADU_ENROLLMENT_STATE_ENROLLED) != ADUC_STATE_STORE_RESULT_OK)
+        {
+            Log_Error("Failed to set enrollment state in state store");
+        }
+    }
+    return oldState;
+}
+
+/**
+ * @brief Sets the correlation id for the enrollment request message.
+ * @param enrollmentData The enrollment data object.
+ * @param correlationId The correlation id to set.
+ */
+void EnrollmentData_SetCorrelationId(ADUC_Enrollment_Request_Operation_Data* enrollmentData, const char* correlationId)
+{
+    if (enrollmentData == NULL || correlationId == NULL)
+    {
+        Log_Error("Null input (enrollmentData:%p, correlationId:%p)", enrollmentData, correlationId);
+        return;
+    }
+
+    strncpy(
+        enrollmentData->enrReqMessageContext.correlationId,
+        correlationId,
+        sizeof(enrollmentData->enrReqMessageContext.correlationId) - 1);
+
+    enrollmentData->enrReqMessageContext
+        .correlationId[sizeof(enrollmentData->enrReqMessageContext.correlationId) - 1] = '\0';
+}
+
+/**
+ * @brief Handles creating side-effects in response to incoming enrollment data from enrollment response.
+ *
+ * @param enrollmentData The enrollment data.
+ * @param isEnrolled The boolean value of enrolled in the message.
+ * @param duInstance The du instance.
+ * @param context The retriable operation context.
+ * @return true on success.
+ */
+bool handle_enrollment_side_effects(
+    ADUC_Enrollment_Request_Operation_Data* enrollmentData,
+    bool isEnrolled,
+    const char* duInstance,
+    ADUC_Retriable_Operation_Context* context)
+{
+    bool succeeded = false;
+    ADUC_STATE_STORE_RESULT stateStoreResult = ADUC_STATE_STORE_RESULT_ERROR;
+
+    EnrollmentData_SetState(
+        enrollmentData,
+        isEnrolled
+            ? ADU_ENROLLMENT_STATE_ENROLLED
+            : ADU_ENROLLMENT_STATE_NOT_ENROLLED,
+        isEnrolled ? "enrolled" : "not enrolled");
```

Comment:
what's this last parameter?  Seems weird to be sending friendly strings to a function.

Reply:
It seems to be used just for a "... (reason:%s)" in EnrollmentData_SetState's Log_Info trace. I wil make it just NULL here that will trace "... (reason:unknown)", which seems alright for now. Likely may remove this const char* reason arg in future PR.